### ISSUE TITLE
Buildings model fails in NFExpression.makeSubscriptedExp

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#12359]
@perost please check [PumpsPrimaryDedicated](https://libraries.openmodelica.org/branches/master/Buildings_latest/files/Buildings_latest_Buildings.Templates.Plants.HeatPumps.Components.Validation.PumpsPrimaryDedicated.err): flattening fails with

```
[Buildings 11.0.1-master/Templates/Components/Data/PumpMultiple.mo:15:10-15:17:writable] Notification: From here:
[Buildings 11.0.1-master/Templates/Components/Data/PumpMultiple.mo:14:3-20:65:writable] Error: Non-array modification '1' for array component 'm_flow_nominal', possibly due to missing 'each'.
```
this is weird, because the 'each' prefix is indeed found in the [referenced source code](https://github.com/lbl-srg/modelica-buildings/blob/16be3880b2bb1804207803bfc3da3359df5f917e/Buildings/Templates/Components/Data/PumpMultiple.mo#L15). Can you please check?

Keeping @AndreaBartolini and @deepak19015 in the loop.

**Comment by perost** [comment]
The issue with the `each` prefix seems to have been fixed in the library, though I'm not sure what the issue actually was. But now the model fails during flattening without any error message instead. Here's an MWE for that issue:
```Modelica
record flowParameters
  parameter Real V_flow[:];
end flowParameters;

record PumpMultiple
  parameter flowParameters fp[2](V_flow = if true then {{0}, {0}} else {{0}});
end PumpMultiple;
```

**Comment by perost** [comment]
#12447 solves the issue that caused the model to fail without an error message, but unfortunately it still fails in the frontend due to some other issue.

**Comment by perost** [comment]
`PumpsPrimaryDedicated` used to fail with the error message:
```
Error: Internal error NFCeval.evalBuiltinAbs got invalid arguments ((#EMPTY#)[3])
```
This is now fixed with #13313. Unfortunately it now fails later with another error instead:
```
[NFFrontEnd/NFExpression.mo:1669:7-1670:100:writable] Error: Internal error NFExpression.makeSubscriptedExp:
too few dimensions in 
Buildings.Templates.Components.Data.PumpSingle(Buildings.Templates.Components.Types.Pump.Multiple, 
({23.90057361376673, 23.90057361376673})[$i1], ({87800.00000000001, 87800.00000000001})[$i1], 
({Buildings.Fluid.Movers.Data.Generic(Buildings.Fluid.Movers.BaseClasses.Characteristics.flowParameters({0.0, 
0.024006538474593588, 0.048013076949187175}, {100092.00000000001, 87800.00000000001, 36876.00000000001}), 
0.06539712205147909, 100092.00000000001, 
Buildings.Fluid.Movers.BaseClasses.Types.HydraulicEfficiencyMethod.EulerNumber, 
Buildings.Fluid.Movers.BaseClasses.Types.MotorEfficiencyMethod.GenericCurve, true, 
Buildings.Fluid.Movers.BaseClasses.Characteristics.efficiencyParameters({0.0}, {0.7}), 
Buildings.Fluid.Movers.BaseClasses.Characteristics.efficiencyParameters({0.0}, {0.7}), 
Buildings.Fluid.Movers.BaseClasses.Characteristics.efficiencyParameters_yMot({0.0}, {0.7}),
Buildings.Fluid.Movers.BaseClasses.Characteristics.powerParameters({0.0}, {0.0}), 
Buildings.Fluid.Movers.BaseClasses.Euler.peak(0.03269856102573954, 72208.13803353971, 0.7), 
Buildings.Fluid.Movers.BaseClasses.Euler.peak(0.03269856102573954, 72208.13803353971, 0.7), true, 2805.312317304277,
0.7, Buildings.Fluid.Movers.BaseClasses.Characteristics.efficiencyParameters_yMot({0.0, 0.1, 0.2, 0.3, 0.4, 0.6, 0.8, 1.0, 1.2},
{7e-7, 0.23457013720271008, 0.43594753193080205, 0.5695606470329043, 0.6490372922266817, 0.7, 0.7, 0.7, 0.7}), true, 1.0, 1.0, {1.0}, true),
Buildings.Fluid.Movers.Data.Generic(Buildings.Fluid.Movers.BaseClasses.Characteristics.flowParameters({0.0,
0.024006538474593588, 0.048013076949187175}, {100092.00000000001, 87800.00000000001, 36876.00000000001}),
0.06539712205147909, 100092.00000000001,
Buildings.Fluid.Movers.BaseClasses.Types.HydraulicEfficiencyMethod.EulerNumber,
Buildings.Fluid.Movers.BaseClasses.Types.MotorEfficiencyMethod.GenericCurve, true,
Buildings.Fluid.Movers.BaseClasses.Characteristics.efficiencyParameters({0.0}, {0.7}),
Buildings.Fluid.Movers.BaseClasses.Characteristics.efficiencyParameters({0.0}, {0.7}),
Buildings.Fluid.Movers.BaseClasses.Characteristics.efficiencyParameters_yMot({0.0}, {0.7}),
Buildings.Fluid.Movers.BaseClasses.Characteristics.powerParameters({0.0}, {0.0}),
Buildings.Fluid.Movers.BaseClasses.Euler.peak(0.03269856102573954, 72208.13803353971, 0.7),
Buildings.Fluid.Movers.BaseClasses.Euler.peak(0.03269856102573954, 72208.13803353971, 0.7), true, 2805.312317304277,
0.7, Buildings.Fluid.Movers.BaseClasses.Characteristics.efficiencyParameters_yMot({0.0, 0.1, 0.2, 0.3, 0.4, 0.6, 0.8, 1.0, 1.2},
{7e-7, 0.23457013720271008, 0.43594753193080205, 0.5695606470329043, 0.6490372922266817, 0.7, 0.7, 0.7, 0.7}), true,
1.0, 1.0, {1.0}, true)})[$i1], 995.586) to apply subscripts [$i1]
```

**Comment by casella** [comment]
Original ticket fixed
